src/ModelContextProtocol/Server/TemporaryAIFunctionFactory.cs

```diff
@@ -568,9 +568,7 @@ static bool IsAsyncMethod(MethodInfo method)
                 throw new ArgumentException("Parameter is missing a name.", nameof(parameter));
             }
 
-            // Resolve the contract used to marshal the value from JSON -- can throw if not supported or not found.
             Type parameterType = parameter.ParameterType;
-            JsonTypeInfo typeInfo = serializerOptions.GetTypeInfo(parameterType);
```

Comment:
Pretty sure the same issue also manifested for return parameters. I take it you dropped those changes intentionally?

Reply:
That was a larger set of changes, it's much less of an issue (as opposed to parameters that are things like IMcpServer and CancellationToken that have no chance of being in the JSO), and this code will (hopefully) all be deleted tomorrow, so I chose to let it be. We can bring it over if you think it's worthwhile, though.

Reply:
It's fine if we wait. I also decided against backporting the fix for the same reason.